apps/theming-designer/src/components/SemanticSlots.tsx

```diff
@@ -432,33 +58,107 @@ export const SemanticSlots: React.StatelessComponent<ISemanticSlotsProps> = (pro
     );
   };
 
-  const fillVariantSlotsList = (variantType: VariantThemeType, mappingType: {}): JSX.Element[] => {
+  const trimPaletteSlots = (paletteSlots: IPalette): IPaletteSlots => {
+    let trimmedPaletteSlots: IPaletteSlots = {};
+    for (let palette in paletteSlots) {
+      if (
+        palette.startsWith('theme') ||
+        palette.startsWith('neutral') ||
+        palette === 'black' ||
+        palette === 'white' ||
+        palette === 'redDark' ||
+        palette === 'accent'
+      ) {
+        trimmedPaletteSlots[palette] = (paletteSlots as any)[palette];
+      }
+    }
+    return trimmedPaletteSlots;
+  };
+
+  const isASemanticColor = (semanticString: string): boolean => {
+    return (
+      semanticString.startsWith('body') ||
+      semanticString.startsWith('primary') ||
+      semanticString.startsWith('disabled') ||
+      semanticString.startsWith('focus') ||
+      semanticString.startsWith('variant') ||
+      semanticString.startsWith('default') ||
+      semanticString.includes('action') ||
+      semanticString.includes('link') ||
+      semanticString.startsWith('accent') ||
+      semanticString.startsWith('list') ||
+      semanticString.startsWith('button') ||
+      semanticString.startsWith('menu') ||
+      semanticString.startsWith('input') ||
+      semanticString.startsWith('small')
+    );
+  };
+
+  const trimSemanticSlotsOrNames = <T extends unknown>(semanticSlots: T) => {
```

Comment:
do you need the `extends unknown`?
also you can say this returns a T as well

Reply:
@aneeshack4, I'm also curious about the usage of `unknown` here because I am unfamiliar with the new TypeScript keyword.

What does `T extends unknown` express?

Reply:
`unknown` is a type-safe alternative to `any` https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type. T must extend something, it can technically extend `any` here too but according to the docs, `unknown` is more type safe... 

When I try to set the return type to `T` errors get thrown saying Type 'any[]' is not assignable to type 'T'.ts for `trimmedSemanticSlotNames` and Type 'ISemanticSlots' is not assignable to type for `trimmedSemanticSlots`

I would like to find a way to specify a return type for this method that would accommodate both `trimmedSemanticSlotNames` and `trimmedSemanticSlots` - that's one of the reasons

Reply:
@cliffkoh showed me how to get rid of all the any usage so no longer using unknown!